woocommerce/woocommerce-ios
# Order Creation: Add order items to order creation API request

Closes: #5410, Closes: #5682

## Description

This adds support for mapping order items in the remote API request for order creation.

It also fixes a couple issues with order creation for products in the UI layer:
* The `Create` button wasn't tappable after adding products to the order.
* Product quantity changes could be lost when the list of items in the order changed. (This was not visually apparent on the new order screen but caused the quantity to be reset to 1, which could be seen after creating the order.)

## Changes

API support:
* Adds `items` case to `CreateOrderField`.
* Adds `items` field handling on Networking layer. This maps the `productID` and `quantity` for each item in the order.
* Adds unit test for mapping the new `items` field.

UI layer fixes:
* Adds `navigationButtonID` in the `NewOrder` view and `ProductsSection` view, to fix the Create button issue.
* Adds a `quantity` parameter to `ProductRowViewModel`, to set a quantity other than 1 for a product row. (That way, when the product rows are updated for the `NewOrder` view, any previous quantity changes are retained.)
* Updates the unit test for product quantity changes to check that the quantity change is retained when the order items change.

## Testing

1. Build and run the app in debug mode (to ensure the feature flag is enabled).
2. Make sure Order Creation is enabled under Settings > Experimental Features.
3. Go to the Orders tab and tap the + button. Tap "Create order" in the bottom sheet.
4. Tap "Add product" and select a product to add it to the order. Change its quantity.
5. Add at least one more product to the order.
6. Tap the "Create" button that appeared in the navigation bar.
7. Confirm Order Details shows the correct products and quantities.

## Screenshots

https://user-images.githubusercontent.com/8658164/146838284-a02200e6-d72f-45e7-877a-bf56f4692562.mp4




## Submitter Checklist

Update release notes:
